patch to dkim.c:6092:49

```
dkim.c:6092:49: warning: size argument in 'strlcpy' call appears to be size 
of the source; expected the size of the destination [-Wstrlcpy-strlcat-size]
        strlcpy((char *) dkim->dkim_zdecode, z, strlen(z));
                                                ~~~~~~~^~
```

Proposed patch:

    -       strlcpy((char *) dkim->dkim_zdecode, z, strlen(z));
    +       strlcpy((char *) dkim->dkim_zdecode, z, strlen(dkim->dkim_zdecode));

--- a/libopendkim/dkim.c
+++ b/libopendkim/dkim.c
@@ -6117,13 +6117,13 @@ dkim_ohdrs(DKIM *dkim, DKIM_SIGINFO *sig, u_char **ptrs, int *pcnt)
 		if (dkim->dkim_zdecode == NULL)
 		{
 			dkim_error(dkim, "unable to allocate %d byte(s)",
-			           strlen(z));
+			           MAXHEADERS);
 			return DKIM_STAT_NORESOURCE;
 		}
 	}
 
 	/* copy it */
-	strlcpy((char *) dkim->dkim_zdecode, z, strlen(z));
+	strlcpy((char *) dkim->dkim_zdecode, z, MAXHEADERS);
 
 	/* decode */
 	for (ch = (u_char *) strtok_r(z, "|", &last);

